core/frontend/src/types/autopilot/parameter-table.ts

```diff
@@ -0,0 +1,110 @@
+
+import { parameters_service } from '../frontend_services'
+import Parameter from './parameter'
+
+const notifier = new Notifier(parameters_service)
+
+// Parameter metadata as in the JSON files
+interface Metadata {
+  Description?: string
+  DisplayName: string
+  Increment?: string
+  Range?: {
+    high: string
+    low: string
+  }
+  RebootRequired?: string,
+  ReadOnly?: string,
+  Bitmask?: {[key:number] : string}
+  Values?: {[key:number] : string}
+  User?: string
+}
+
+interface MetadataCategory {
+  [key: string]: Metadata | number; // number deals with a sneaky {"json": 0} entry
+}
+interface MetadataFile {
+    [key: string]: MetadataCategory;
+}
+export default class ParametersTable {
+    parametersDict: {[key: number] : Parameter} = {}
+
+    count = 0
+
+    metadata = {} as Dictionary<Metadata>
+
+    constructor() {
+      this.fetchMetadata()
+    }
+
+    fetchMetadata(): void {
+      if (autopilot.vehicle_type === null) {
+        // Check again later if we have a vehicle type identified
+        setTimeout(() => { this.fetchMetadata() }, 1000)
+        return
+      }
+      // default to submarine
+      let metadata = ardusub_metadata as MetadataFile
+      // This is to avoid importing a 40 lines enum from mavlink and adding a switch case with 40 cases
+      if (autopilot.vehicle_type.toLowerCase().includes('copter')
+      || autopilot.vehicle_type.toLowerCase().includes('rotor')) {
+        metadata = arducopter_metadata
+      } else if (autopilot.vehicle_type.toLowerCase().includes('rover')
+      || autopilot.vehicle_type.toLowerCase().includes('boat')) {
+        metadata = ardurover_metadata
+      }
+
+      for (const category of Object.values(metadata)) {
+        for (const [name, parameter] of Object.entries(category)) {
+          if (isNumber(parameter)) { // ignore "json" entry
+            console.log(`ignoring ${name} : ${parameter}`)
+            continue
+          }
+          this.metadata[name] = parameter
+        }
+      }
+    }
+
+    addParam(param: Parameter): void {
+      if (param.name in this.metadata) {
+        param.setDescription(this.metadata[param.name].Description ?? '')
+        param.setShortDescription(this.metadata[param.name].DisplayName)
+        const {
+          Values, Bitmask, ReadOnly, Increment, RebootRequired,
+        } = this.metadata[param.name]
+        param.setOptions(Values)
+        param.setBitmask(Bitmask)
+        param.setReadOnly(ReadOnly === 'True')
+        param.setIncrement(Increment ? parseFloat(Increment) : undefined)
+        param.setRebootRequired(RebootRequired === 'True')
+      }
+      this.parametersDict[param.id] = param
+    }
+
+    updateParam(param_name: string, param_value: number): void {
+      const index = Object.entries(this.parametersDict).find(([key, value]) => value.name === param_name)
+      if (!index) {
+        const message = `unable to set parameter ${param_name}. Parameter not found`
+        console.warn(message)
```

Comment:
I believe that I was not clear, but the `console.` call is done already by `notifier.pushError`, so it's unnecessary since it'll display the message twice.